libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -20,6 +23,35 @@ void VotePacketHandler::validatePacketRlpFormat([[maybe_unused]] const PacketDat
   }
 }
 
+bool VotePacketHandler::isVoteRoundStepRelevant(const std::shared_ptr<Vote> &vote) const {
+  auto checking_round = pbft_mgr_->getPbftRound();
+  // If period is not the same we assuming current round is equal to 0
+  // So we won't accept votes for future period with round bigger than kVoteAcceptingSteps
+  if (pbft_chain_->getPbftChainSize() != vote->getPeriod()) {
```

Comment:
I think this is wrong logic, you should be checking pbft_chain_->getPbftChainSize() + 1, but dont use chainSize,  use:

const auto [checking_round, current_pbft_period] = pbft_mgr_->getPbftRoundAndPeriod();
if (current_pbft_period != vote->getPeriod()) {
  checking_round = 0
}